controllers/backup/reconciler.go

```diff
@@ -0,0 +1,557 @@
+			for name := range currentCluster {
+				delete(clusterMap, name)
+			}
+
+			cmDataMap[common.AerospikeClustersKey] = clusterMap
+		}
+	}
+
+	if routineIface, ok := cmDataMap[common.BackupRoutinesKey]; ok {
+		if routineMap, ok := routineIface.(map[string]interface{}); ok {
+			currentRoutines := backupDataMap[common.BackupRoutinesKey].(map[string]interface{})
+			for name := range currentRoutines {
+				delete(routineMap, name)
+			}
+
+			cmDataMap[common.BackupRoutinesKey] = routineMap
+		}
+	}
+
+	updatedConfig, err := yaml.Marshal(cmDataMap)
+	if err != nil {
+		return err
+	}
+
+	cm.Data[common.BackupServiceConfigYAML] = string(updatedConfig)
+
+	if err := r.Client.Update(
+		context.TODO(), cm, common.UpdateOption,
+	); err != nil {
+		return fmt.Errorf(
+			"failed to update Backup Service ConfigMap: %v",
+			err,
+		)
+	}
+
+	r.Log.Info("Removed Backup info from existing ConfigMap",
+		"name", r.aeroBackup.Spec.BackupService.Name,
+		"namespace", r.aeroBackup.Spec.BackupService.Namespace,
+	)
+
+	return nil
+}
+
+func (r *SingleBackupReconciler) scheduleOnDemandBackup() error {
+	// There can be only one on-demand backup allowed right now.
+	if len(r.aeroBackup.Status.OnDemandBackups) > 0 &&
+		r.aeroBackup.Spec.OnDemandBackups[0].ID == r.aeroBackup.Status.OnDemandBackups[0].ID {
+		r.Log.Info("On-demand backup already scheduled for the same ID",
+			"ID", r.aeroBackup.Status.OnDemandBackups[0].ID)
+		return nil
+	}
+
+	r.Log.Info("Schedule on-demand backup",
+		"ID", r.aeroBackup.Spec.OnDemandBackups[0].ID, "routine", r.aeroBackup.Spec.OnDemandBackups[0].RoutineName)
+
+	backupServiceClient, err := backup_service.GetBackupServiceClient(r.Client, &r.aeroBackup.Spec.BackupService)
+	if err != nil {
+		return err
+	}
+
+	if err = backupServiceClient.ScheduleBackup(r.aeroBackup.Spec.OnDemandBackups[0].RoutineName,
+		r.aeroBackup.Spec.OnDemandBackups[0].Delay); err != nil {
+		r.Log.Error(err, "Failed to schedule on-demand backup")
+		return err
+	}
+
+	r.Log.Info("Scheduled on-demand backup", "ID", r.aeroBackup.Spec.OnDemandBackups[0].ID,
+		"routine", r.aeroBackup.Spec.OnDemandBackups[0].RoutineName)
+
+	return nil
+}
+
+func (r *SingleBackupReconciler) reconcileBackup() error {
+	if err := r.reconcileScheduledBackup(); err != nil {
+		return err
+	}
+
+	return r.reconcileOnDemandBackup()
+}
+
+func (r *SingleBackupReconciler) reconcileScheduledBackup() error {
+	specHash, err := utils.GetHash(string(r.aeroBackup.Spec.Config.Raw))
+	if err != nil {
+		return err
+	}
+
+	statusHash, err := utils.GetHash(string(r.aeroBackup.Status.Config.Raw))
+	if err != nil {
+		return err
+	}
+
+	if specHash == statusHash {
+		r.Log.Info("Backup config not changed")
+		return nil
+	}
+
+	r.Log.Info("Registering backup")
+
+	serviceClient, err := backup_service.GetBackupServiceClient(r.Client, &r.aeroBackup.Spec.BackupService)
+	if err != nil {
+		return err
+	}
+
+	config, err := serviceClient.GetBackupServiceConfig()
```

Comment:
can we totally rely either on status config or config from backupservice. as of now few things are compared with status and few things with backupservice config.

Reply:
Changed as per our discussion